### DISCUSSION TITLE
How to show code completion + docs while writing?

### DISCUSSION THREAD

**Discussion by ghost** [helix-editor/helix#3038] [Q&A] [answered]
I didn't understood how to show the docs while writing, like this image:   

![](https://github.com/helix-editor/helix/raw/master/screenshot.png)

I can see all functions and fields, but the docs doesn't show. I only know how to see the docs using `space + k`.

**Comment by sudormrfbin** [comment]
Selecting a completion item should automatically show it's documentation in a popup as in the image. What language are you working with? Which language server?

**Reply by ghost** [reply]
I'm using Typescript, with the `typescript-language-server`. I'm on Termux, though.  

![example](https://files.catbox.moe/39ouen.jpg)
![](https://files.catbox.moe/4q1gl3.jpg)

**Comment by ghost** [comment, marked as answer]
Aparently this just doesn't work on typescript/javascript, i just created a .dart file and the docs shows automatically.

**Reply by the-mikedavis** [reply]
The typescript language server may need a call to resolve the completion request [`completionItem/resolve`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_completion)

https://github.com/helix-editor/helix/issues/2052
